hasjob/models/jobpost_alert.py

```diff
@@ -0,0 +1,99 @@
+# -*- coding: utf-8 -*-
+
+from datetime import datetime, timedelta
+from coaster.sqlalchemy import StateManager
+from ..utils import random_long_key
+from . import db, BaseMixin, LabeledEnum, User, AnonUser
+
+__all__ = ['JobPostSubscription', 'JobPostAlert', 'jobpost_alert_table']
+
+
+class EMAIL_FREQUENCY(LabeledEnum):
+    DAILY = (1, 'Daily')
+    WEEKLY = (7, 'Weekly')
+
+
+class JobPostSubscription(BaseMixin, db.Model):
+    __tablename__ = 'jobpost_subscription'
+
+    filterset_id = db.Column(None, db.ForeignKey('filterset.id'), nullable=False)
+    filterset = db.relationship('Filterset', backref=db.backref('subscriptions', lazy='dynamic'))
+    email = db.Column(db.Unicode(254), nullable=True)
+    user_id = db.Column(None, db.ForeignKey('user.id'), nullable=True, index=True)
+    user = db.relationship(User)
+    anon_user_id = db.Column(None, db.ForeignKey('anon_user.id'), nullable=True, index=True)
+    anon_user = db.relationship(AnonUser)
+
+    active = db.Column(db.Boolean, nullable=False, default=False, index=True)
+    email_verify_key = db.Column(db.String(40), nullable=True, default=random_long_key, unique=True)
+    unsubscribe_key = db.Column(db.String(40), nullable=True, default=random_long_key, unique=True)
+    subscribed_at = db.Column(db.DateTime, nullable=False, default=db.func.utcnow())
+    email_verified_at = db.Column(db.DateTime, nullable=True, index=True)
+    unsubscribed_at = db.Column(db.DateTime, nullable=True)
+
+    _email_frequency = db.Column('email_frequency',
+        db.Integer, StateManager.check_constraint('email_frequency', EMAIL_FREQUENCY),
+        default=EMAIL_FREQUENCY.DAILY, nullable=True)
+    email_frequency = StateManager('_email_frequency', EMAIL_FREQUENCY, doc="Email frequency")
+
+    __table_args__ = (db.UniqueConstraint('filterset_id', 'email'),
+        db.CheckConstraint(
+            db.case([(user_id != None, 1)], else_=0) + db.case([(anon_user_id != None, 1)], else_=0) <= 1,  # NOQA
```

Comment:
I used `<=1` since anon user tracking isn't enabled right now, so there's a case where both `g.user` and `g.anon_user` will be `None` but we'll have the user's email id.

Reply:
Let's fix anon user handling then. This has to be `== 1`.